core/src/main/scala/org/apache/spark/SSLOptions.scala

```diff
@@ -22,6 +22,23 @@ import java.io.File
 import com.typesafe.config.{Config, ConfigFactory, ConfigValueFactory}
 import org.eclipse.jetty.util.ssl.SslContextFactory
 
+/** SSLOptions class is a common container for SSL configuration options. It offers methods to
+  * generate specific objects to configure SSL for different communication protocols.
+  *
+  * SSLOptions is intended to provide the maximum common set of SSL settings, which are supported
+  * by the protocol, which it can generate the configuration for. Since Akka doesn't support client
+  * authentication with SSL, SSLOptions cannot support it either.
+  *
+  * @param enabled             enables or disables SSL; if it is set to false, the rest of the
+  *                            settings are disregarded
+  * @param keyStore            a path to the key-store file
+  * @param keyStorePassword    a password to access the key-store file
+  * @param keyPassword         a password to access the private key in the key-store
+  * @param trustStore          a path to the trust-store file
+  * @param trustStorePassword  a password to access the trust-store file
+  * @param protocol            SSL protocol (remember that SSLv3 was compromised) supported by Java
+  * @param enabledAlgorithms   a set of encryption algorithms to use
+  */
 private[spark] case class SSLOptions(
```

Comment:
Are you sure we want defaults on all fields of that case class? Feels counter-intuitive to security in general.